tests/test_optimizer_secagg.py

```diff
@@ -0,0 +1,380 @@
+        self,
+    ) -> Tuple[Dict[str, Any], Optional[Dict[str, Any]]]:
+        raise NotImplementedError("This module does not support SecAgg.")
+
+
+def generate_scaffold_aux_var(n_feats: int = 32) -> Dict[str, AuxVar]:
+    """Generate auxiliary variables from a Scaffold Optimizer.
+
+    Args:
+        n_feats: Dimensionality of features of the mocked linear
+            model. Parameters will have `n_feats + 1` size due
+            to the addition of a bias.
+
+    Use mock (random-valued) model weights and input gradients,
+    and return auxiliary variables after running a single local
+    step.
+    """
+    # Ensure no GPU is used.
+    declearn.utils.set_device_policy(gpu=False)
+    # Run a mock training step using a Scaffold optimizer.
+    gradients = declearn.model.torch.TorchVector(
+        {
+            "kernel": torch.clamp(torch.randn((n_feats, 1)), -2.5, 2.5),
+            "bias": torch.clamp(torch.randn((1,)), -2.5, 2.5),
+        }
+    )
+    weights = declearn.model.torch.TorchVector(
+        {
+            "kernel": torch.clamp(torch.randn((n_feats, 1)), 5.0, 5.0),
+            "bias": torch.clamp(torch.randn((1,)), 5.0, 5.0),
+        }
+    )
+    optimizer = Optimizer(lr=0.001, modules=[ScaffoldClientModule()])
+    optimizer.step(gradients, weights)
+    # Return the resulting auxiliary variables.
+    return optimizer.get_aux()
+
+
+
+class TestFlattenAuxVarForSecagg(unittest.TestCase):
+    """Unit tests for 'flatten_auxvar_for_secagg'."""
+
+    @staticmethod
+    def assert_flattened_auxvar_types(
+        aux_var: Dict[str, AuxVar],
+        flattened: List[float],
+        enc_specs: List[ValueSpec],
+        cleartext: List[Optional[Dict[str, Any]]],
+        clear_cls: List[Tuple[str, Type[AuxVar]]],
+    ) -> None:
+        """Verify that flattened aux var match documented specifications."""
+        # Assert that 'flattened' is a list of float values.
+        assert isinstance(flattened, list)
+        assert all(isinstance(x, float) for x in flattened)
+        # Assert that 'enc_specs' lists module-wise specifications.
+        assert isinstance(enc_specs, list)
+        assert len(enc_specs) == len(aux_var)
+        for specs in enc_specs:
+            assert isinstance(specs, list)
+            assert all(isinstance(s, tuple) and (len(s) == 3) for s in specs)
+        # Assert that 'cleartext' is a list of module-wise optional dicts.
+        assert isinstance(cleartext, list)
+        assert all(isinstance(x, (dict, type(None))) for x in cleartext)
+        assert len(cleartext) == len(aux_var)
+        # Assert that 'clear_cls' is a list of module-wise (name, type) tuples.
+        assert isinstance(clear_cls, list)
+        assert len(clear_cls) == len(aux_var)
+        assert clear_cls == [(key, type(val)) for key, val in aux_var.items()]
+
+    def test_flatten_auxvar_simple(self) -> None:
+        """Test flattening a dict with a single 'SimpleAuxVar' instance."""
```

Comment:
these unit tests are great, buit they don't respect the fbm conventions for unit tests: expecially, method name should be numbered